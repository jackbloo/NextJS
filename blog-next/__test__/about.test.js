import { render, screen } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import About from '../pages/about'

describe('Unit Testing of About', () => {
  it('Should renders about component', async () => {
      await act(async () =>{
        await render(<About data={[]}/>)
      })

    const heading = screen.getByTestId('about-container')
    expect(heading).toMatchSnapshot();
  })
})