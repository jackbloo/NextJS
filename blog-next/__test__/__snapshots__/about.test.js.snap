// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing of About Should renders about component 1`] = `
<div
  class="container"
  data-testid="about-container"
>
  <main
    class="main"
  >
    <div
      class="navbar"
      data-testid="navbar-container"
    >
      <div
        class="rightNavbar"
      >
        <a
          href="/"
        >
          Digitsense Blog
        </a>
      </div>
      <div
        class="leftNavbar"
      >
        <div
          class="nav"
          style="font-weight: normal; text-decoration: none;"
        >
          <a
            href="/"
          >
            Home
          </a>
        </div>
        <div
          class="nav"
          style="font-weight: bold; text-decoration: underline;"
        >
          <a
            href="/about"
          >
            About us
          </a>
        </div>
        <div
          class="nav"
          style="font-weight: normal; text-decoration: none;"
        >
          <a
            href="/undefined"
          />
        </div>
      </div>
    </div>
    <h1
      class="title"
    >
      About this blog
    </h1>
    <div
      class="profileContainer"
    >
      <div
        class="profile"
        data-testid="profile-container"
      >
        <div
          class="profilePicture"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 250px; height: 300px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <img
              alt="Minimalist"
              data-nimg="fixed"
              decoding="async"
              src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1519085360753-af0119f7cbe7%3Fixlib%3Drb-1.2.1%26ixid%3DMnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8%26auto%3Dformat%26fit%3Dcrop%26w%3D687%26q%3D80&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1519085360753-af0119f7cbe7%3Fixlib%3Drb-1.2.1%26ixid%3DMnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8%26auto%3Dformat%26fit%3Dcrop%26w%3D687%26q%3D80&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1519085360753-af0119f7cbe7%3Fixlib%3Drb-1.2.1%26ixid%3DMnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8%26auto%3Dformat%26fit%3Dcrop%26w%3D687%26q%3D80&w=640&q=75 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
        <h1>
          Harry Maguire
        </h1>
        <p>
          Chief Executive Officer (CEO)
        </p>
        <p
          class="text"
        >
           Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Praesent elementum facilisis leo vel fringilla est ullamcorper eget. At imperdiet dui accumsan sit amet nulla facilisi morbi tempus.
        </p>
      </div>
      <div
        class="profile"
        data-testid="profile-container"
      >
        <div
          class="profilePicture"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 250px; height: 300px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <img
              alt="Minimalist"
              data-nimg="fixed"
              decoding="async"
              src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fflagged%2Fphoto-1553642618-de0381320ff3%3Fixlib%3Drb-1.2.1%26ixid%3DMnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8%26auto%3Dformat%26fit%3Dcrop%26w%3D687%26q%3D80&w=640&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fflagged%2Fphoto-1553642618-de0381320ff3%3Fixlib%3Drb-1.2.1%26ixid%3DMnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8%26auto%3Dformat%26fit%3Dcrop%26w%3D687%26q%3D80&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fflagged%2Fphoto-1553642618-de0381320ff3%3Fixlib%3Drb-1.2.1%26ixid%3DMnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8%26auto%3Dformat%26fit%3Dcrop%26w%3D687%26q%3D80&w=640&q=75 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
        <h1>
          Jadon Sancho
        </h1>
        <p>
          Chief Technology Officer (CTO)
        </p>
        <p
          class="text"
        >
           Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Praesent elementum facilisis leo vel fringilla est ullamcorper eget. At imperdiet dui accumsan sit amet nulla facilisi morbi tempus.
        </p>
      </div>
    </div>
  </main>
  <footer
    class="footer"
    data-testid="footer-container"
  >
    <a
      href="https://digit-sense.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Powered by
       
      <span
        class="logo"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2772%27%20height=%2740%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Vercel Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=%2Fdigitsense.jpeg&w=256&q=75"
            srcset="/_next/image?url=%2Fdigitsense.jpeg&w=96&q=75 1x, /_next/image?url=%2Fdigitsense.jpeg&w=256&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </span>
    </a>
  </footer>
</div>
`;
